test-utils/archunit/src/main/java/org/camunda/bpm/engine/test/archunit/ArchRulesTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.test.archunit;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.junit.ArchUnitRunner;
+import com.tngtech.archunit.lang.ArchRule;
+import org.junit.runner.RunWith;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+
+@RunWith(ArchUnitRunner.class)
+@AnalyzeClasses(
+    importOptions = {
+        ImportOption.DoNotIncludeJars.class,
+        ImportOption.DoNotIncludeTests.class
+    },
+    packages = "org.camunda.bpm"
+)
+public class ArchRulesTest {
+
+  @ArchTest
+  public static final ArchRule public_api_should_not_expose_implementation = classes().that()
```

Comment:
I read that ArchUnit supports generating more readable names from snake case rule names ([docs](https://www.archunit.org/userguide/html/000_Index.html#_generating_display_names)). However, I think the `junit.displayName.replaceUnderscoresBySpaces` property has to be enabled for that. Do we use that property somewhere outside the code? I could not spot it in the PR.
If we are not using it and not planing to do so, should we stick with the conventional camel case pattern?

Reply:
Good point. Let's revert it back to camel case. The ArchUnit code examples just distracted me. It's better to be consistent with our coding style guide.

Reply:
Feel free to merge afterwards :)
```suggestion
  public static final ArchRule publicApiShouldNotExposeImplementation = classes().that()
```